pallets/stateful-storage/src/rpc/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+// Strong Documentation Lints
+#![deny(
+	rustdoc::broken_intra_doc_links,
+	rustdoc::missing_crate_level_docs,
+	rustdoc::invalid_codeblock_attributes,
+	missing_docs
+)]
+
+//! Custom APIs for [Stateful-Storage](../pallet_stateful_storage/index.html)
+
+use common_primitives::{
+	msa::MessageSourceId,
+	schema::*,
+	stateful_storage::{ItemizedStoragePageResponse, PaginatedStorageResponse},
+};
+use jsonrpsee::{
+	core::{async_trait, Error as RpcError, RpcResult},
+	proc_macros::rpc,
+	types::error::{CallError, ErrorCode, ErrorObject},
+};
+use pallet_stateful_storage_runtime_api::StatefulStorageRuntimeApi;
+use sp_api::{ApiError, ProvideRuntimeApi};
+use sp_blockchain::HeaderBackend;
+use sp_runtime::{generic::BlockId, traits::Block as BlockT, DispatchError};
+use sp_std::vec::Vec;
+use std::sync::Arc;
+
+#[cfg(test)]
+mod tests;
+
+/// Frequency Stateful Storage Custom RPC API
+#[rpc(client, server)]
+pub trait StatefulStorageApi<BlockHash> {
```

Comment:
I doubt it, but would it be possible to fold this into the messages rpc? Not just add the endpoints, but add the responses to the existing `messages.getBySchemaId` endpoint?

Reply:
there are a number of challenges with trying to use the same rpc for getting messages and storages.
- Messages are schema oriented, which means we only need schema but storages are msa_id and schema oriented.
- Messages response types are very different from storage since we have pagination on them but storages right not are not paginated since the assumption is that they can not grow that big (might be a wrong assumption)
- Rust would not allow us to return different types unless we change the rpc to return an enum and define those different response types in there.

We can always unify them after merge since that would need to be a separate new endpoint and existing message rpc endpoint would not support it without changes which would break backwards compatibility.